pkg/plugins/runtime/k8s/controllers/inbound_converter.go

```diff
@@ -189,18 +189,26 @@ func ProtocolTagFor(svc *kube_core.Service, svcPort *kube_core.ServicePort) stri
 
 	if svcPort.AppProtocol != nil {
 		protocolValue = *svcPort.AppProtocol
-	} else {
-		protocolValue = svc.Annotations[protocolAnnotation]
+		// `appProtocol` can be any protocol and if we don't explicitly support
+		// it, let the default below take effect
+		if core_mesh.ParseProtocol(protocolValue) == core_mesh.ProtocolUnknown {
+			protocolValue = ""
+		}
+	}
+
+	if explicitKumaProtocol, ok := svc.Annotations[protocolAnnotation]; ok && protocolValue == "" {
```

Comment:
arguably this annotation should _always_ override but that'd be a "breaking" change